core/src/main/scala/com/salesforce/op/ModelInsights.scala

```diff
@@ -557,6 +558,25 @@ case object ModelInsights {
             case _ => None
           }
           val keptIndex = indexInToIndexKept.get(h.index)
+          val featureStd = getIfExists(h.index, s.featuresStatistics.variance).getOrElse(1.0)
+          val sparkFtrContrib = keptIndex
+            .map(i => contributions.map(_.applyOrElse(i, (_: Int) => 0.0))).getOrElse(Seq.empty)
+          val LRStandardization = checkLRStandardization(model).getOrElse(false)
+          val labelStd = label.distribution.getOrElse(1.0) match {
+            case Continuous(_, _, _, variance) => math.sqrt(variance)
+            // for (binary) logistic regression we only need to multiply by feature standard deviation
+            case Discrete(domain, prob) =>
+              def computeVariance(domain: Seq[String], prob: Seq[Double]): Double = {
```

Comment:
you dont need the inner function and you can do it faster. simply do:
```scala
case Discrete(domain, prob) =>
   val (weighted, sqweighted) = (domain zip prob).foldLeft((0.0, 0.0)) { case ((weightSum, sqweightSum), (d, p)) =>
         val floatD = d.toDouble
         val weight = floatD * prob
         val sqweight = floatD * weight
         (weightSum+ weight, sqweightSum + sqweight)  
   }
   sqweighted - weighted
```

Reply:
another question, since `domain` is `Seq[String]` how can we be sure that `d.toDouble` wont throw an error?! should we handle it gracefully?

Reply:
For regression problem where there are too few unique labels (and they get treated as categoricals), `d.toDouble` should work fine. I'm not sure how it will behave on classification. For classification I assume that `domain = Array("0", "1")`, is this correct ?

Reply:
Why did we decide to define `domain` as `Seq[String]` in the first place? @Jauntbox @leahmcguire

Reply:
If you have the raw label rather than the indexed label you will get strings - this was designed to support that

Reply:
is it safe to do `toDouble` in this case ?

Reply:
Any value which is not numeric with throw an exception, e.g. `"".toDouble`

Reply:
but will there ever be non numeric string in the label field...?